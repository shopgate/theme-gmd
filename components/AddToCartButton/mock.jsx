/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
