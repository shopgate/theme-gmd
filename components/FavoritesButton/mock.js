/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
