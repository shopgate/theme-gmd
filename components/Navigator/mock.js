import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const router = {
  routing: false,
  stack: [
    {
      id: '44a34767-2683-4414-9de4-fd736808fdab',
      params: {},
      pathname: '/',
      pattern: '/',
      query: {},
      state: {},
    },
  ],
};

export const defaultState = {
  navigator: {
    backgroundColor: 'red',
    enabled: true,
    filterOpen: false,
    filterAttributeOpen: false,
    loginOpen: false,
    searchActive: false,
    showCartIcon: true,
    showIconShadow: false,
    showProgressBar: true,
    showSearch: true,
    showTitle: true,
    textColor: 'white',
    viewTracking: false,
  },
  search: {
    suggestion: {},
  },
  router,
  view: {
    isLoading: false,
  },
  ui: {
    general: {
      title: null,
    },
  },
};

/**
 * Creates store.
 * @param {Object} props Props which would override state.navigator props.
 * @return {Object}
 */
export const getStore = props => configureStore([thunk])({
  ...defaultState,
  navigator: {
    ...defaultState.navigator,
    ...props,
  },
  search: {
    suggestions: {},
  },
});
