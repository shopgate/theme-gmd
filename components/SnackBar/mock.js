/**
 * Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked toast state
 * @type {{toast: [null,null]}}
 */
export const mockedState = {
  toast: [
    {
      id: 1,
      message: 'Lorem ipsum dolor sit amet.',
      duration: 6000,
    },
  ],
};
