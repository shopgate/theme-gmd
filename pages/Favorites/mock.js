/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

