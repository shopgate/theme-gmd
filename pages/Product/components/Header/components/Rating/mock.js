/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
