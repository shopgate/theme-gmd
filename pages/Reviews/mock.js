/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
