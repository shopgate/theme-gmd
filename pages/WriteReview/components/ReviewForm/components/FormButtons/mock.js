/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
