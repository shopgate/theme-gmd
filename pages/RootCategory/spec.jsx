import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MockedView } from 'Components/View/mock';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { initialCategoryState, categoryState } from '@shopgate/pwa-common-commerce/category/mock';

const mockedView = MockedView;
const mockedStore = configureStore();
jest.mock('Components/View', () => mockedView);
let store;

const results = [
  [{
    type: 'NAVIGATE',
    action: 'PUSH',
    location: '/category/6d656e',
    state: { title: 'Men' },
  }],
];

/**
 * Creates component
 * @param {boolean} state State that would be used for store.
 * @return {ReactWrapper}
 */
const createComponent = (state) => {
  store = mockedStore(state);
  /* eslint-disable global-require */
  const RootCategory = require('./index').default;
  /* eslint-enable global-require */
  return mount(
    <Provider store={store} ><RootCategory /></Provider>,
    mockRenderOptions
  );
};

describe('<RootCategory> page', () => {
  beforeEach(() => {
    jest.resetModules();
  });
  it('should render', () => {
    const component = createComponent(categoryState);
    expect(component).toMatchSnapshot();
    expect(component.find('CategoryList').length).toEqual(1);
    expect(component.find('ListItem').length).toEqual(2);
  });

  it('should render empty', () => {
    const component = createComponent(initialCategoryState);
    expect(component).toMatchSnapshot();
    expect(component.find('CategoryList').length).toEqual(0);
    expect(component.find('ListItem').length).toEqual(0);
  });

  it('should navigate to subcategory', () => {
    const wrapper = createComponent(categoryState);

    wrapper.find('CategoryList').find('Connect(Link)').at(0).simulate('click');
    wrapper.update();
    expect(store.getActions()).toEqual(results[0]);
  });
});
