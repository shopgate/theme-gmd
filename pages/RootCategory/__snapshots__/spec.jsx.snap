// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RootCategory> page should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(shouldUpdate(RootCategory))>
    <shouldUpdate(RootCategory)
      categories={
        Array [
          Object {
            "children": Array [],
            "childrenCount": 0,
            "childrenSort": "import",
            "description": "",
            "expires": 9999999999999,
            "externalUrl": "http://material-fashion.myshopify.com/collections/men",
            "id": "men",
            "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346609@2x.jpg",
            "isFetching": false,
            "name": "Men",
            "parent": Object {
              "id": null,
              "name": "Default- category",
            },
            "path": "Men",
            "productCount": 6,
            "sort": 0,
          },
          Object {
            "children": Array [],
            "childrenCount": 1,
            "childrenSort": "import",
            "description": "",
            "expires": 9999999999999,
            "externalUrl": "http://material-fashion.myshopify.com/collections/women",
            "id": "women",
            "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346610@2x.jpg",
            "isFetching": false,
            "name": "Women",
            "parent": Object {
              "id": null,
              "name": "Default- category",
            },
            "path": "Women",
            "productCount": 6,
            "sort": 0,
          },
        ]
      }
      dispatch={[Function]}
    >
      <RootCategory
        categories={
          Array [
            Object {
              "children": Array [],
              "childrenCount": 0,
              "childrenSort": "import",
              "description": "",
              "expires": 9999999999999,
              "externalUrl": "http://material-fashion.myshopify.com/collections/men",
              "id": "men",
              "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346609@2x.jpg",
              "isFetching": false,
              "name": "Men",
              "parent": Object {
                "id": null,
                "name": "Default- category",
              },
              "path": "Men",
              "productCount": 6,
              "sort": 0,
            },
            Object {
              "children": Array [],
              "childrenCount": 1,
              "childrenSort": "import",
              "description": "",
              "expires": 9999999999999,
              "externalUrl": "http://material-fashion.myshopify.com/collections/women",
              "id": "women",
              "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346610@2x.jpg",
              "isFetching": false,
              "name": "Women",
              "parent": Object {
                "id": null,
                "name": "Default- category",
              },
              "path": "Women",
              "productCount": 6,
              "sort": 0,
            },
          ]
        }
        dispatch={[Function]}
      >
        <MockedView>
          <div>
            <CategoryList
              categories={
                Array [
                  Object {
                    "children": Array [],
                    "childrenCount": 0,
                    "childrenSort": "import",
                    "description": "",
                    "expires": 9999999999999,
                    "externalUrl": "http://material-fashion.myshopify.com/collections/men",
                    "id": "men",
                    "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346609@2x.jpg",
                    "isFetching": false,
                    "name": "Men",
                    "parent": Object {
                      "id": null,
                      "name": "Default- category",
                    },
                    "path": "Men",
                    "productCount": 6,
                    "sort": 0,
                  },
                  Object {
                    "children": Array [],
                    "childrenCount": 1,
                    "childrenSort": "import",
                    "description": "",
                    "expires": 9999999999999,
                    "externalUrl": "http://material-fashion.myshopify.com/collections/women",
                    "id": "women",
                    "imageUrl": "https://shopgate-public.s3.amazonaws.com/categories/24453/1346610@2x.jpg",
                    "isFetching": false,
                    "name": "Women",
                    "parent": Object {
                      "id": null,
                      "name": "Default- category",
                    },
                    "path": "Women",
                    "productCount": 6,
                    "sort": 0,
                  },
                ]
              }
            >
              <List>
                <List
                  className=""
                >
                  <ul
                    className="css-gkw1j5 "
                  >
                    <ListItem
                      className="css-1brrprg css-1ym3l2n"
                      isSelected={false}
                      key="child-0/.$men"
                    >
                      <li
                        className="css-1p5q8vq css-1brrprg css-1ym3l2n css-16gj3o1"
                      >
                        <div
                          className="css-1nnz83b"
                        >
                          <Portal
                            key="men"
                            name="category.item"
                            props={
                              Object {
                                "categoryId": "men",
                              }
                            }
                          >
                            <Item
                              className={null}
                              image={null}
                              isDisabled={false}
                              isSelected={false}
                              link="/category/6d656e"
                              onClick={null}
                              rightComponent={null}
                              testId={null}
                              title="Men"
                            >
                              <Glow
                                className={null}
                                color="#f7f7f7"
                                styles={
                                  Object {
                                    "container": null,
                                    "glow": null,
                                  }
                                }
                              >
                                <div
                                  aria-hidden={true}
                                  className="css-13azwyo"
                                  onTouchTap={[Function]}
                                  style={null}
                                >
                                  <div
                                    className="css-1bmd5dz"
                                    style={Object {}}
                                  />
                                  <Connect(Link)
                                    href="/category/6d656e"
                                    onClick={null}
                                    state={
                                      Object {
                                        "title": "Men",
                                      }
                                    }
                                  >
                                    <Link
                                      className=""
                                      disabled={false}
                                      href="/category/6d656e"
                                      navigate={[Function]}
                                      onClick={null}
                                      replace={false}
                                      state={
                                        Object {
                                          "title": "Men",
                                        }
                                      }
                                      tag="div"
                                    >
                                      <div
                                        aria-hidden={true}
                                        className="css-c6ki83 "
                                        onClick={[Function]}
                                        role="link"
                                      >
                                        <Grid
                                          className="css-1qcr7hz"
                                          component="div"
                                          wrap={false}
                                        >
                                          <div
                                            className="css-1qcr7hz css-h7r55a"
                                          >
                                            <div
                                              className="css-jwxytc"
                                            />
                                            <GridItem
                                              className="css-10fxwvh"
                                              component="div"
                                              grow={1}
                                              shrink={1}
                                            >
                                              <div
                                                className="css-10fxwvh css-1gs0ko2"
                                              >
                                                Men
                                              </div>
                                            </GridItem>
                                          </div>
                                        </Grid>
                                      </div>
                                    </Link>
                                  </Connect(Link)>
                                </div>
                              </Glow>
                            </Item>
                          </Portal>
                        </div>
                      </li>
                    </ListItem>
                    <ListItem
                      className="css-1brrprg"
                      isSelected={false}
                      key="child-1/.$women"
                    >
                      <li
                        className="css-1p5q8vq css-1brrprg css-16gj3o1"
                      >
                        <div
                          className="css-1nnz83b"
                        >
                          <Portal
                            key="women"
                            name="category.item"
                            props={
                              Object {
                                "categoryId": "women",
                              }
                            }
                          >
                            <Item
                              className={null}
                              image={null}
                              isDisabled={false}
                              isSelected={false}
                              link="/category/776f6d656e"
                              onClick={null}
                              rightComponent={null}
                              testId={null}
                              title="Women"
                            >
                              <Glow
                                className={null}
                                color="#f7f7f7"
                                styles={
                                  Object {
                                    "container": null,
                                    "glow": null,
                                  }
                                }
                              >
                                <div
                                  aria-hidden={true}
                                  className="css-13azwyo"
                                  onTouchTap={[Function]}
                                  style={null}
                                >
                                  <div
                                    className="css-1bmd5dz"
                                    style={Object {}}
                                  />
                                  <Connect(Link)
                                    href="/category/776f6d656e"
                                    onClick={null}
                                    state={
                                      Object {
                                        "title": "Women",
                                      }
                                    }
                                  >
                                    <Link
                                      className=""
                                      disabled={false}
                                      href="/category/776f6d656e"
                                      navigate={[Function]}
                                      onClick={null}
                                      replace={false}
                                      state={
                                        Object {
                                          "title": "Women",
                                        }
                                      }
                                      tag="div"
                                    >
                                      <div
                                        aria-hidden={true}
                                        className="css-c6ki83 "
                                        onClick={[Function]}
                                        role="link"
                                      >
                                        <Grid
                                          className="css-1qcr7hz"
                                          component="div"
                                          wrap={false}
                                        >
                                          <div
                                            className="css-1qcr7hz css-h7r55a"
                                          >
                                            <div
                                              className="css-jwxytc"
                                            />
                                            <GridItem
                                              className="css-10fxwvh"
                                              component="div"
                                              grow={1}
                                              shrink={1}
                                            >
                                              <div
                                                className="css-10fxwvh css-1gs0ko2"
                                              >
                                                Women
                                              </div>
                                            </GridItem>
                                          </div>
                                        </Grid>
                                      </div>
                                    </Link>
                                  </Connect(Link)>
                                </div>
                              </Glow>
                            </Item>
                          </Portal>
                        </div>
                      </li>
                    </ListItem>
                  </ul>
                </List>
              </List>
            </CategoryList>
          </div>
        </MockedView>
      </RootCategory>
    </shouldUpdate(RootCategory)>
  </Connect(shouldUpdate(RootCategory))>
</Provider>
`;

exports[`<RootCategory> page should render empty 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(shouldUpdate(RootCategory))>
    <shouldUpdate(RootCategory)
      categories={null}
      dispatch={[Function]}
    >
      <RootCategory
        categories={null}
        dispatch={[Function]}
      >
        <MockedView>
          <div />
        </MockedView>
      </RootCategory>
    </shouldUpdate(RootCategory)>
  </Connect(shouldUpdate(RootCategory))>
</Provider>
`;
